readthedocs/projects/signals.py

```diff
@@ -27,26 +28,11 @@ def handle_project_import(sender, **kwargs):
     project = sender
     request = kwargs.get('request')
 
-    for provider in ['github', 'bitbucket']:
-        if provider in project.repo:
-            session = oauth_utils.get_oauth_session(user=request.user, provider=provider)
-            if not session:
-                break
-            if provider == 'github':
-                try:
-                    resp = oauth_utils.add_github_webhook(session, project)
-                    if resp.status_code == 201:
-                        messages.success(request, _('GitHub webhook activated'))
-                # pylint: disable=bare-except
-                # TODO this should be audited for exception types
-                except:
-                    log.exception('GitHub Hook creation failed', exc_info=True)
-            elif provider == 'bitbucket':
-                try:
-                    resp = oauth_utils.add_bitbucket_webhook(session, project)
-                    if resp.status_code == 200:
-                        messages.success(request, _('BitBucket webhook activated'))
-                # pylint: disable=bare-except
-                # TODO this should be audited for exception types
-                except:
-                    log.exception('BitBucket Hook creation failed', exc_info=True)
+    for service_cls in registry:
```

Comment:
Yeah, seems we left it in signals for extending in readthedocs.com. I think a dynamic class based view would work better here